src/Nethermind/Nethermind.Evm/EvmPooledMemory.cs

```diff
@@ -257,18 +266,22 @@ public void Dispose()
             }
         }
 
-        private static UInt256 MaxInt32 = (UInt256)int.MaxValue;
-
         public static long Div32Ceiling(in UInt256 length)
         {
-            UInt256 rem = length & 31;
-            UInt256 result = length >> 5;
-            if (!rem.IsZero)
+            if ((length.u1 | length.u2 | length.u2) != 0)
```

Comment:
`lenght.u2` duplicated, did you mean `lenght.u3`. 

Also you are introducing a lot of `uint256.u1 | uint256.u2 | uint256.u3` code - maybe an extension method for that or method inside the UInt256 class? (can be aggressively inlined if needed)

Reply:
Yep, also added the extension